### PR TITLE
Fix multiple issues with type annotations

### PR DESCRIPTION
**Pull request by ghost**
- added tests that showcase the problems
- fixed the problems

These changes would allow me to remove many "# type: ignore" comments in my codebase.

### CODE REVIEW

**Comment by mruberry** [comment]
@gchanan Who do we have to review type hint related PRs?

File: torch/cuda/__init__.pyi [hunk 1]
```diff
@@ -1,5 +1,9 @@
 from typing import Optional, Tuple, Union
 from .. import device as _device
+from .random import get_rng_state as get_rng_state, get_rng_state_all as get_rng_state_all, \
+    set_rng_state as set_rng_state, set_rng_state_all as set_rng_state_all, manual_seed as manual_seed, \
+    manual_seed_all as manual_seed_all, seed as seed, seed_all as seed_all, initial_seed as initial_seed
```
Comment (bartosz319) [inline torch/cuda/__init__.pyi:5]:
```suggestion
from .random import *
```

This would probably work too, but I haven't tested it

**Comment by bartosz319** [comment]
@ezyang perhaps you could take a look if you have time?

**Comment by ezyang** [comment]
Thanks for this contribution @bartosz319. Do you mind if we wait for #36584 to go in first before landing this one?

**Comment by ezyang** [comment]
cc @rgommers

File: torch/nn/modules/container.pyi.in [hunk 2]
```diff
@@ -10,6 +10,7 @@ class Container(Module):
 
 
 T = TypeVar('T')
+ModuleT = TypeVar('ModuleT', bound=Module)
```
Comment (ezyang) [inline torch/nn/modules/container.pyi.in:13]: This is related to #35566. This is actually a pretty clever workaround, maybe we should start applying it everywhere.

**Review by ezyang — APPROVED** [review]
This looks great

**Comment by bartosz319** [comment]
> Thanks for this contribution @bartosz319. Do you mind if we wait for #36584 to go in first before landing this one?

Thanks for the review. Sure, we can wait. Let me know if any changes to my PR are needed in response to the other PR being merged.

**Review by rgommers — COMMENTED** [review]
Thanks @bartosz319, this looks great. Could you please rebase it (because gh-36584 was merged)? Rebasing with `pytorchbot` didn't work, perhaps because you sent your PR from master rather than a feature branch.

File: test/type_hint_tests/module_list.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
+from typing import Iterable
+
+import torch
+
+# ModuleList with elements of a specific type
+class FooModule(torch.nn.Module):
+    def ten(self):
+        # type: () -> int
```
[the remark this answers is not in the document]
Comment (ezyang) [reply test/type_hint_tests/module_list.py:8]: Let's switch to inline annotations!

**Comment by bartosz319** [comment]
I think I've addressed all review comments now

**Comment by bartosz319** [comment]
@ezyang What are the internal failures? Is it something I can fix?

**Comment by ezyang** [comment]
the internal failures are all fake. Gonna retry land

**Pull request closed without merging**